appveyor.yml

```diff
@@ -1,31 +1,67 @@
-image: Visual Studio 2019
-
 shallow_clone: true
 
 platform:
     - x64
 
 environment:
     matrix:
-        - PYTHON_VERSION: 3.7
+        - APPVEYOR_BUILD_WORKER_IMAGE: Visual Studio 2019
           MINICONDA_DIRNAME: C:\Miniconda37-x64
-        - PYTHON_VERSION: 3.8
+          PYTHON_VERSION: 3.7
+        - APPVEYOR_BUILD_WORKER_IMAGE: Visual Studio 2019
+          MINICONDA_DIRNAME: C:\Miniconda38-x64
+          PYTHON_VERSION: 3.8
+        - APPVEYOR_BUILD_WORKER_IMAGE: Visual Studio 2019
           MINICONDA_DIRNAME: C:\Miniconda38-x64
+          PYTHON_VERSION: 3.9
+        - APPVEYOR_BUILD_WORKER_IMAGE: Ubuntu
+          PYTHON_VERSION: 3.7
+        - APPVEYOR_BUILD_WORKER_IMAGE: Ubuntu
+          PYTHON_VERSION: 3.8
+        - APPVEYOR_BUILD_WORKER_IMAGE: Ubuntu
+          PYTHON_VERSION: 3.9
+        - APPVEYOR_BUILD_WORKER_IMAGE: macOS
+          PYTHON_VERSION: 3.7
+        - APPVEYOR_BUILD_WORKER_IMAGE: macOS
+          PYTHON_VERSION: 3.8
+        - APPVEYOR_BUILD_WORKER_IMAGE: macOS
+          PYTHON_VERSION: 3.9
 
     ANACONDA_TOKEN:
         secure: $(anaconda_token)
 
+init:
+    - cmd: set "PATH=%MINICONDA_DIRNAME%;%MINICONDA_DIRNAME%\\Scripts;%PATH%"
+    - sh: if [[ "$APPVEYOR_BUILD_WORKER_IMAGE" == "Ubuntu" ]]; then
+              OS=Linux ;
+          else
+              OS=MacOSX ;
+          fi ;
+          curl -sL https://repo.anaconda.com/miniconda/Miniconda3-latest-$OS-x86_64.sh > miniconda.sh
+    - sh: bash miniconda.sh -b -p $HOME/miniconda;
+    - sh: export PATH="$HOME/miniconda/bin:$HOME/miniconda/lib:$PATH";
+
 install:
-    - set "PATH=%MINICONDA_DIRNAME%;%MINICONDA_DIRNAME%\\Scripts;%PATH%"
-    - conda config --set always_yes yes
+    - conda config --set always_yes yes --set changeps1 no
     - conda update -q conda
     - conda env create -f environment.yml
     - activate cadquery
-    - conda install -c conda-forge -c defaults -c cadquery python=%PYTHON_VERSION%
+    - cmd: conda install -c conda-forge -c defaults -c cadquery cadquery=master python=%PYTHON_VERSION%
+    - sh: conda install -c conda-forge -c defaults -c cadquery cadquery=master python=$PYTHON_VERSION
```

Comment:
Doesn't this install both `cadquery=master` and `python=3.9`? I don't think we want to install CQ from Conda here.

Reply:
My understanding is that it will install the current master branch of CQ and will use Python 3.9 for the environment. The goal is to test the latest CQ with a specific Python version.

Reply:
I think I see your point. The Action already checked out CadQuery's repo so this isn't needed, correct?

Reply:
This workflow is being run on every PR, right (I'm not sure where AppVeyor gets those instructions from)? 

We don't want to accidentally run any tests on the Conda version instead of the PR version. Or if we dropped something essential from `environment.yml` in a PR, this line would still install every dependency from master and tests would still pass.